### DISCUSSION TITLE
Combine vs AsyncStreams

### DISCUSSION THREAD

**Discussion by jaroslavas** [pointfreeco/swift-composable-architecture#2257] [General]
From what I see in the code of this library, Combine-based interfaces are deprecated. What I can't find is the reason behind that. Is there a public discussion or an explanation of why this was done? 
For a background, I'm thinking if using Combine in a TCA-based app is a good idea or is going all-in on AsyncStreams is the way to go?

**Comment by mbrandonw** [comment]
The use of Combine in TCA is not deprecated, just the publisher-specific operators on `Effect`. You are always free to create an effect from a publisher via this static function: 

https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/effectpublisher/publisher(_:)

**Reply by yermukhanbet** [reply]
hey @mbrandonw , was the documentation moved somewhere else? The link you sent is not found :(

**Reply by kamcma** [reply]
https://pointfreeco.github.io/swift-composable-architecture/1.2.0/documentation/composablearchitecture/effect/publisher(_:)
